packages/react-component-library/src/components/Select/Select.tsx

```diff
@@ -38,6 +53,112 @@ export const Select = ({
 
   const isControlled = value !== undefined
 
+  // Only used when isMulti is true
+  const [selectedItems, setSelectedItems] = useState<string[]>(() => {
+    if (isMulti && initialValue) {
+      return props.initialValue as string[] // Safe cast as when isMulti is true initialValue must be string[]
+    }
+    return []
+  })
+  const currentSelectedItems = useMemo(() => {
+    if (!isMulti) {
+      return []
+    }
+
+    return isControlled && Array.isArray(value) ? value : selectedItems
+  }, [isMulti, isControlled, value, selectedItems])
+
+  const callMultiOnChange = useCallback(
+    (newValue: string[]) => {
+      const handler = onChange as MultiSelectProps['onChange']
+      handler?.(newValue)
+    },
+    [onChange]
+  )
+
+  const callSingleOnChange = useCallback(
+    (newValue: string | null) => {
+      const handler = onChange as SingleSelectProps['onChange']
+      handler?.(newValue)
+    },
+    [onChange]
+  )
+
+  const multipleSelection = useMultipleSelection({
+    selectedItems: isMulti ? currentSelectedItems : [],
+    onStateChange({ selectedItems: newSelectedItems, type }) {
+      switch (type) {
```

Comment:
How about something like this instead of these switch statements:

https://github.com/Royal-Navy/design-system/commit/c9815cb93d7820089e16e240e94186c598f1d1f0